src/bindings/cpp/tests/tests.hpp

```diff
@@ -37,8 +39,147 @@ using namespace kdb;
+{
+	for (auto key : keys)
+	{
+		stream << key;
+	}
+	return stream;
+}
+
+/**
+ * @brief This function checks if the meta key set of two keys are equal.
+ *
+ * @param key1 This parameter stores the first meta key set this function uses for comparison.
+ * @param key2 This parameter stores the second meta key set this function uses for comparison.
+ *
+ * @retval true if all meta keys of `key1` and `key2` are equal
+ * @retval false otherwise
+ */
+bool isMetaDataEqual (kdb::Key & key1, kdb::Key & key2)
+{
+	key1.rewindMeta ();
+	key2.rewindMeta ();
+
+	while (key1.nextMeta ())
+	{
+		key2.nextMeta ();
+		if (!key2.currentMeta ()) return false;
+		if (key1.currentMeta ().getName () != key2.currentMeta ().getName ()) return false;
+		if (key1.currentMeta ().getString () != key2.currentMeta ().getString ()) return false;
+	}
+
+	return key1.nextMeta () == key2.nextMeta ();
+}
+
+/**
+ * @brief This function checks if two keys are equal.
+ *
+ * @param key1 This parameter stores the first key this function compares.
+ * @param key2 This parameter stores the second key this function compares.
+ *
+ * @retval true if all attributes of `key1` and `key2` are equal
+ * @retval false otherwise
+ */
+bool isKeyEqual (kdb::Key & key1, kdb::Key & key2)
+{
+	if (key1.getName () != key2.getName ()) return false;
+	if ((key1.isString () && key2.isBinary ()) || (key1.isBinary () && key2.isString ())) return false;
+	if (key1.getBinarySize () != key2.getBinarySize ()) return false;
+	if (memcmp (key1.getValue (), key2.getValue (), key1.getBinarySize ())) return false;
+	return isMetaDataEqual (key1, key2);
+}
+
+/**
+ * @brief This function checks if two key sets are equal.
+ *
+ * @param keys1 This parameter stores the first key set this function compares.
+ * @param keys2 This parameter stores the second key set this function compares.
+ *
+ * @retval true if all keys of `keys1` and `keys2` are equal
+ * @retval false otherwise
+ */
+bool isKeySetEqual (kdb::KeySet & keys1, kdb::KeySet & keys2)
+{
+	if (keys1.size () != keys2.size ()) return false;
+
+	keys1.rewind ();
+	keys2.rewind ();
+	while (keys1.next ())
+	{
+		keys2.next ();
+		if (!keys2.current ()) return false;
+		kdb::Key key1 = keys1.current ();
+		kdb::Key key2 = keys2.current ();
+		if (!isKeyEqual (key1, key2)) return false;
+	}
+
+	return keys1.next () == keys2.next ();
+}
+
+/**
+ * @brief This function compares the given key sets.
+ *
+ * @param keys1 This parameter stores the first key set this function compares.
+ * @param keys2 This parameter stores the second key set this function compares.
+ *
+ * @retval AssertionSuccess if all keys of `keys1` and `keys2` are equal
+ * @retval AssertionFailure otherwise
+ */
+testing::AssertionResult compareKeySet (kdb::KeySet & keys1, kdb::KeySet & keys2)
+{
+	if (isKeySetEqual (keys1, keys2))
+		return testing::AssertionSuccess ();
+	else
+		return testing::AssertionFailure () << endl
```

Comment:
One endl at the end should be enough, otherwise `\n` is more appropriate (does not flush the stream).